src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -352,6 +352,15 @@ void InferRequest::SetBlob(const std::string& name, const Blob::Ptr& data) {
                         if (batch_idx >= 0)
                             SetBatch(blobDesc.getDims()[batch_idx]);
                     }
```

Comment:
[random spot] The original intent of having 2 impls of InferRequest is that it's really hard to combine and maintain legacy features (such as dynamic batch) with dynamic shapes. And new infer request impl should NOT have any dynamic batch stuff. At this moment it's not true. Moreover, current version doesn't have some changes from POC branch, and the impl won't even work for dynamic flow.

So please do the following:
1. Cleanup new infer request impl from legacy freatures
2. Ensure that new impl has proper handling for dynamic shapes case

Reply:
Cleaned up legacy features(dynamic batch, preprocessing) from new infer request. and by applying this change to POC branch, I also confirmed that it works well w/o runtime error in dynamic scenario with BERT base model(cased).